packages/redirects-utils/src/index.js

```diff
@@ -0,0 +1,88 @@
+const fs = require('fs')
+const path = require('path')
+const { promisify } = require('util')
+
+const writeFileAsync = promisify(fs.writeFile)
+
+const resolveConfig = require('@netlify/config')
+const { getConfigPath } = require('@netlify/config')
+const { parseRedirectsFormat, parseNetlifyConfig } = require('netlify-redirect-parser')
+const configorama = require('configorama')
+
+async function parseFile(parser, name, filePath) {
+  const result = await parser(filePath)
+  if (result.errors.length) {
+    console.error(`Warnings while parsing ${name} file:`)
+    result.errors.forEach(err => console.error(`  ${err.lineNum}: ${err.line} -- ${err.reason}`))
+  }
+  return result.success
+}
+
+async function getAll(projectPath) {
+  const rulesUnique = new Set()
+
+  const configPath = await getConfigPath(undefined, projectPath)
+  const redirectsFilePath = path.resolve(projectPath, '_redirects')
+
+  if (fs.existsSync(redirectsFilePath)) {
+    const fileName = redirectsFilePath
+      .split(path.sep)
+      .pop();
+
+    (await parseFile(parseRedirectsFormat, fileName, redirectsFilePath))
+      .forEach(r => rulesUnique.add(JSON.stringify(r)))
+  }
+  if (fs.existsSync(configPath)) {
+    const fileName = configPath
+      .split(path.sep)
+      .pop();
+
+    (await parseFile(parseNetlifyConfig, fileName, configPath))
+      .forEach(r => rulesUnique.add(JSON.stringify(r)))
+  }
+
+  return Array.from(rulesUnique).map(item => JSON.parse(item))
+}
+
+async function remove(rule = {}, projectPath) {
+  const rules = (await getAll(projectPath)).filter(
+    r =>
+      !(
+        r.from === rule.from &&
+        r.to === rule.to &&
+        r.status === rule.status &&
+        r.force === rule.force &&
+        r.query === rule.query &&
+        JSON.stringify(r.conditions) === JSON.stringify(rule.conditions) &&
+        JSON.stringify(r.headers) === JSON.stringify(rule.headers) &&
+        r.signed === rule.signed
+      ),
+  )
+
+  const configPath = await getConfigPath(undefined, projectPath)
+  const config = await resolveConfig(configPath, { cwd: projectPath })
```

Comment:
Utils are going to be included in core. This means re-parsing the config might not be required. 

I suppose other plugins could be adding new values tho... 

Figuring out a performant way to allow adding/remove/modifying redirects in memory might make sense as a longer term solution. We don't need to do it not but eventually some kind of state might be a nice optimization

Reply:
Yes I can confirm utils are now included by default. If you rebase to `master`, you should see [how utils are directly included now](https://github.com/netlify/build/blob/master/packages/build/src/plugins/child/utils.js).

In that code, you can pass the `netlifyConfig` directly to the core util, i.e. you don't need to re-parse it.